cmd/manager/main.go

```diff
@@ -4,140 +4,152 @@ import (
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 )
 
-const (
-	initialWait                = time.Duration(1) * time.Minute
-	updateNotificationSendWait = time.Duration(10) * time.Minute
-)
+// TODO(tflannag): Should we be setting up a simple pprof server for profiling?
+// TODO(tflannag): We should avoid using the global logrus as our logger
+// TODO(tflannag): Can we remove the exit() function and just log on error level -> os.Exit(1)?
+// TODO(tflannag): Why are we vendoring in the CatalogSource type? Why can't we just rely on whatever we get from the operator-framework/api dependency?
+// TODO(tflannag): We should probably make the readiness/liveness probe(s) port configurable
+// TODO(tflannag): Double-check we haven't regressed with exposing Prometheus metrics
+// TODO(tflannag): May need to add unit tests for that useTLS functions
+// TODO(tflannag): We should aim towards cleaning up this main() implementation so it's not just a giant function
+// TODO(tflannag): We should avoid having to maintain a global list of mappings for the default CatalogSource(s)
+// TODO(tflannag): What's the openshift recommendation on leader election? Do we want to avoid using the current leader for life implementation?
 
-var (
-	version     = flag.Bool("version", false, "displays marketplace source commit info.")
-	tlsKeyPath  = flag.String("tls-key", "", "Path to use for private key (requires tls-cert)")
-	tlsCertPath = flag.String("tls-cert", "", "Path to use for certificate (requires tls-key)")
+const (
+	leaderElectionConfigMapName = "marketplace-operator-lock"
 )
 
 func printVersion() {
-	log.Printf("Go Version: %s", runtime.Version())
-	log.Printf("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH)
-	log.Printf("operator-sdk Version: %v", sdkVersion.Version)
+	logrus.Printf("Go Version: %s", runtime.Version())
+	logrus.Printf("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH)
+	logrus.Printf("operator-sdk Version: %v", sdkVersion.Version)
 }
 
 func main() {
 	printVersion()
 
-	// Parse the command line arguments
-	flag.StringVar(&defaults.Dir, "defaultsDir",
-		"", "the directory where the default CatalogSources are stored")
-	var clusterOperatorName string
+	var (
+		clusterOperatorName string
+		tlsKeyPath          string
+		tlsCertPath         string
+		version             bool
+	)
 	flag.StringVar(&clusterOperatorName, "clusterOperatorName", "", "the name of the OpenShift ClusterOperator that should reflect this operator's status, or the empty string to disable ClusterOperator updates")
+	flag.StringVar(&defaults.Dir, "defaultsDir", "", "the directory where the default CatalogSources are stored")
+	flag.BoolVar(&version, "version", false, "displays marketplace source commit info.")
+	flag.StringVar(&tlsKeyPath, "tls-key", "", "Path to use for private key (requires tls-cert)")
+	flag.StringVar(&tlsCertPath, "tls-cert", "", "Path to use for certificate (requires tls-key)")
```

Comment:
Minor change: just moved these from global variables to the inside the scope of the main function.